ush/extractvars_tools.sh

```diff
@@ -0,0 +1,54 @@
+#! /usr/bin/env bash
+
+gen_parmlist() {
+  infileg=$1
+  requestedvar_in_file1=$2
+  varlist=$3
+  rm -vf "${requestedvar_in_file1}"
+  while read -r vari; do
+    varinfile=$(${WGRIB2} "${infileg}" | grep "${vari}") || true
+    if [[ -n "${varinfile}" ]];then # if varinfile is not empty
+      echo "${vari}" >> "${requestedvar_in_file1}"
+    fi
+  done <"${varlist}"
```

Comment:
I think that `wgrib2` only needs to be read once here.  This would save you some processing time, especially if running on a single core.  Suggestion below is untested.

```suggestion
  file_vars=$(${WGRIB2} "${infileg}")
  while read -r vari; do
    [[ ${file_vars} =~ "${vari}" ]] && echo "${vari}" >> "${requestedvar_in_file1}"
  done <"${varlist}"
```

Reply:
@DavidHuber-NOAA Thank you for finding a significantly more efficient way to perform this function. I agree that `wgrib2` only needs to be executed once to read a grib2 file in this situation. I tried your suggestion and got an `exit 1` error, but only when I source `preamble.sh`. Your suggestion returns `exit 0` when I do not source `preamble.sh`. I tried a similar method as yours and it does not return any errors when I source preamble. Is it ok if I use the following?

```
  file_vars=$(${WGRIB2} "${infileg}")
  while read -r vari; do                                                                                                                                                                      
   if [[ "${file_vars}" =~ "${vari}" && -n "${vari}" ]]; then                                                                                                                                 
     echo "${vari}" >> "${requestedvar_in_file1}"
   fi
  done <"${varlist}" 

```

Reply:
Yeah, I'm good with that way as well.  It's more readable and the double quotes around `file_vars` is good practice.

Reply:
Thanks. This change has been added.